product_contract/models/sale_order.py

```diff
@@ -2,86 +2,76 @@
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
-from odoo import _, fields, api, models
+from odoo import _, api, fields, models
 from odoo.exceptions import ValidationError
 
 
 class SaleOrder(models.Model):
-    _inherit = 'sale.order'
+    _inherit = "sale.order"
 
-    is_contract = fields.Boolean(
-        string='Is a contract', compute='_compute_is_contract'
-    )
-    contract_count = fields.Integer(compute='_compute_contract_count')
-    need_contract_creation = fields.Boolean(
-        compute='_compute_need_contract_creation'
-    )
+    is_contract = fields.Boolean(string="Is a contract", compute="_compute_is_contract")
+    contract_count = fields.Integer(compute="_compute_contract_count")
+    need_contract_creation = fields.Boolean(compute="_compute_need_contract_creation")
 
-    @api.constrains('state')
+    @api.constrains("state")
     def check_contact_is_not_terminated(self):
         for rec in self:
-            if rec.state not in (
-                'sale',
-                'done',
-                'cancel',
-            ) and rec.order_line.filtered('contract_id.is_terminated'):
+            if rec.state not in ("sale", "done", "cancel",) and rec.order_line.filtered(
+                "contract_id.is_terminated"
+            ):
                 raise ValidationError(
                     _("You can't upsell or downsell a terminated contract")
                 )
 
-    @api.depends('order_line.contract_id', 'state')
+    @api.depends("order_line.contract_id", "state")
     def _compute_need_contract_creation(self):
         for rec in self:
-            if rec.state in ('sale', 'done'):
+            rec.need_contract_creation = False
+            if rec.state in ("sale", "done"):
                 line_to_create_contract = rec.order_line.filtered(
                     lambda r: not r.contract_id and r.product_id.is_contract
                 )
                 line_to_update_contract = rec.order_line.filtered(
                     lambda r: r.contract_id
                     and r.product_id.is_contract
                     and r
-                    not in r.contract_id.contract_line_ids.mapped(
-                        'sale_order_line_id'
-                    )
+                    not in r.contract_id.contract_line_ids.mapped("sale_order_line_id")
                 )
                 if line_to_create_contract or line_to_update_contract:
                     rec.need_contract_creation = True
 
-    @api.depends('order_line')
+    @api.depends("order_line")
     def _compute_is_contract(self):
-        self.is_contract = any(self.order_line.mapped('is_contract'))
+        self.is_contract = any(self.order_line.mapped("is_contract"))
 
-    @api.multi
     def _prepare_contract_value(self, contract_template):
         self.ensure_one()
         return {
-            'name': '{template_name}: {sale_name}'.format(
+            "name": "{template_name}: {sale_name}".format(
```

Comment:
Please  use %() or f-strings

Reply:
Problem with f-strings is that they are not supported in 3.5, which might be annoying for some project.
Otherwise, I'm really wondering why you would prefer the old-style formatting -- besides the fact that it's the most used in Odoo, which is not a great argument by itself.

Reply:
>-- besides the fact that it's the most used in Odoo, which is not a great argument by itself.

:+1: Completely agree

Reply:
Thanks a lot, I was not aware of this :+1: